applications/sdp/mspi/src/main.c

```diff
@@ -52,215 +57,490 @@ typedef struct __packed {
 	struct mspi_xfer_packet packet;
 } nrfe_mspi_xfer_packet_t;
 
-struct mspi_config {
-	uint8_t *data;
-	uint8_t data_len;
-	uint8_t word_size;
+enum base_io_mode {
+	BASE_IO_MODE_SINGLE,
+	BASE_IO_MODE_DUAL,
+	BASE_IO_MODE_QUAD,
+	BASE_IO_MODE_OCTAL
 };
 
-struct mspi_dev_config {
-	enum mspi_io_mode io_mode;
-	enum mspi_ce_polarity ce_polarity;
-	uint32_t read_cmd;
-	uint32_t write_cmd;
-	uint8_t cmd_length;  /* Command length in bits. */
-	uint8_t addr_length; /* Address length in bits. */
+struct xfer_io_mode_cfg {
+	enum base_io_mode command;
+	enum base_io_mode address;
+	enum base_io_mode data;
 };
 
-static struct mspi_dev_config mspi_dev_configs;
-
-uint32_t data_buffer[MAX_DATA_LEN + 2];
-
-volatile struct hrt_ll_xfer xfer_ll_params = {
-	.counter_top = 4,
-	.word_size = 4,
-	.data_to_send = NULL,
-	.data_len = 0,
-	.ce_hold = false,
-	.ce_enable_state = false,
+/*************************************************
+ * 	      LOCAL VARIABLES
+ *************************************************/
+
+static const uint8_t pin_to_vio_map[] = {
+	4,  /* Physical pin 0 */
+	0,  /* Physical pin 1 */
+	1,  /* Physical pin 2 */
+	3,  /* Physical pin 3 */
+	2,  /* Physical pin 4 */
+	5,  /* Physical pin 5 */
+	6,  /* Physical pin 6 */
+	7,  /* Physical pin 7 */
+	8,  /* Physical pin 8 */
+	9,  /* Physical pin 9 */
+	10, /* Physical pin 10 */
 };
 
+static volatile uint8_t ce_vios_count;
+static volatile uint8_t ce_vios[CE_PINS_MAX];
+static volatile uint8_t data_vios_count;
+static volatile uint8_t data_vios[DATA_PINS_MAX];
+static volatile struct mspi_cfg nrfe_mspi_cfg;
+static volatile struct mspi_dev_cfg nrfe_mspi_dev_cfg;
+static volatile struct mspi_xfer nrfe_mspi_xfer;
+static volatile struct mspi_xfer_packet nrfe_mspi_xfer_packet;
+static volatile struct hrt_ll_xfer xfer_ll_params;
+
 static struct ipc_ept ep;
 static atomic_t ipc_atomic_sem = ATOMIC_INIT(0);
 
-static void process_packet(const void *data, size_t len, void *priv);
+/*************************************************
+ * 	    GLOBAL FUNCTION PROTOTYPES
+ *************************************************/
 
-static void ep_bound(void *priv)
+__attribute__((interrupt)) void hrt_handler_write(void);
+
+int main(void);
```

Comment:
What is it for? I don't think it is needed.

Reply:
I wanted to create organized sections:
- global functions prototypes
- local functions prototypes
- global function implementations
- local function implementations

And this function was fit for this section, true it is not required but for me it makes code more readable because it lists all functions in short space

Reply:
normaly this function would be in file main.h but it does not exist

Reply:
IMO, in header file should be placed only global functions declarations, not static ones.
Also function declarations should be specified only for functions that require it. It is not required for all functions.

Reply:
ok, removed